File: javascript-examples/v2/indexer_asset_search.js
// Demonstrate an Indexer Asset Search with pagination 

const algosdk = require('algosdk');
const baseServer = "https://testnet-algorand.api.purestake.io/idx2";
const port = "";

const token = {
    'X-API-key' : 'B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab',
}

let indexerClient = new algosdk.Indexer(token, baseServer, port);

(async()=> {

    let i =5;

    let assetInfo = await indexerClient.searchForAssets().limit(1).name('test').do()
    console.log(assetInfo)

    while(i && ('next-token' in assetInfo)) {
        assetInfo = await indexerClient.searchForAssets().limit(1).name('test').nextToken(assetInfo['next-token']).do();
        i--;
        console.log(assetInfo) // Array of assets object
        console.log(assetInfo.assets[0]) // An asset returned
    }

})().catch(e => {
    console.log(e);
});

File: java-examples/IndexerAssetSearch.java
package com.purestake.algosdk.example;

import com.algorand.algosdk.v2.client.common.IndexerClient;

public class IndexerAssetSearch {

    public static void main(String[] args) throws Exception {
        final String IDX_API_ADDR = "https://testnet-algorand.api.purestake.io/idx2";
        final int  IDX_PORT = 443;
        final String IDX_API_TOKEN_KEY = "X-API-Key";
        final String IDX_API_TOKEN = "B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab";

        IndexerClient client = new IndexerClient(IDX_API_ADDR, IDX_PORT, IDX_API_TOKEN, IDX_API_TOKEN_KEY);
        
        String name = "test";
        Long limit = 1L;
        String response = client.searchForAssets().name(name).limit(limit).execute().toString();
        System.out.println(response);
    }
}
